pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -39,114 +39,104 @@ func TestInjectOne(t *testing.T) {
+	secretsCl := injector.client.Core().Secrets(bindings[0].Namespace)
+	secret, err := secretsCl.Get(bindings[0].Name)
 	if err != nil {
 		t.Fatalf("Error when getting secret: %s", err)
 	}
-	testCredentialsInjected(t, secret.Data, cred0)
+	testCredentialsInjected(t, secret.Data, creds[0])
 
-	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
-	secret, err = secretsCl.Get(binding1.Name)
+	secretsCl = injector.client.Core().Secrets(bindings[1].Namespace)
+	secret, err = secretsCl.Get(bindings[1].Name)
 	if err != nil {
 		t.Fatalf("Error when getting secret: %s", err)
 	}
-	testCredentialsInjected(t, secret.Data, cred1)
+	testCredentialsInjected(t, secret.Data, creds[1])
 }
 
 func TestUninjectOne(t *testing.T) {
-	binding, _ := getBindings()
-	cred, _ := getCreds()
+	binding := createBindings()[0]
+	cred := createCreds()[0]
 
 	injector := fakeK8sBindingInjector()
 	inject(t, injector, binding, cred)
 	injector.Uninject(binding)
 
-	secretsCl := injector.client.Core().Secrets(binding.Namespace)
-	secret, err := secretsCl.Get(binding.Name)
-	if err == nil {
-		testCredentialsUninjected(t, secret.Data)
-	}
+	testCredentialsUninjected(t, injector, binding)
 }
 
 func TestUninjectTwo(t *testing.T) {
-	binding0, binding1 := getBindings()
-	cred0, cred1 := getCreds()
+	bindings := createBindings()
+	creds := createCreds()
 
 	injector := fakeK8sBindingInjector()
-	inject(t, injector, binding0, cred0)
-	inject(t, injector, binding1, cred1)
+	inject(t, injector, bindings[0], creds[0])
+	inject(t, injector, bindings[1], creds[1])
 
-	injector.Uninject(binding0)
+	injector.Uninject(bindings[0])
 
-	// test that binding0 is gone
-	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
-	secret, err := secretsCl.Get(binding0.Name)
-	if err == nil {
-		testCredentialsUninjected(t, secret.Data)
-	}
+	// test that bindings[0] is gone
+	testCredentialsUninjected(t, injector, bindings[0])
 
-	//test that binding1 is still there
-	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
-	secret, err = secretsCl.Get(binding1.Name)
+	//test that bindings[1] is still there
+	secretsCl := injector.client.Core().Secrets(bindings[1].Namespace)
+	secret, err := secretsCl.Get(bindings[1].Name)
 	if err != nil {
 		t.Fatalf("Error when getting secret: %s", err)
 	}
-	testCredentialsInjected(t, secret.Data, cred1)
+	testCredentialsInjected(t, secret.Data, creds[1])
 
-	// test that binding1 is gone after uninject
-	injector.Uninject(binding1)
+	// test that bindings[1] is gone after uninject
+	injector.Uninject(bindings[1])
 
-	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
-	secret, err = secretsCl.Get(binding1.Name)
-	if err == nil {
-		testCredentialsUninjected(t, secret.Data)
-	}
+	testCredentialsUninjected(t, injector, bindings[1])
 }
 
-func fakeK8sBindingInjector() *k8sBindingInjector {
-	return &k8sBindingInjector{
-		client: fake.NewSimpleClientset(),
+func createBindings() []*servicecatalog.Binding {
```

Comment:
I was kind of assuming the caller would pass in the # they wanted so that you wouldn't be limited to just 2. And if a new test wanted, say, 10 the code here wouldn't need to change.  But its up to you - its no biggie.